src/vss_tools/vspec/model.py

```diff
@@ -85,13 +85,21 @@ def as_dict(
         self,
         with_extra_attributes: bool = True,
         exclude_fields: list[str] = EXPORT_EXCLUDE_ATTRIBUTES,
+        extended_attributes: tuple[str, ...] = (),
     ) -> dict[str, Any]:
         excludes = exclude_fields.copy()
         if not with_extra_attributes:
-            excludes.extend(self.get_extra_attributes())
-        data = {
-            k: v for k, v in self.model_dump(mode="json", exclude_none=True, exclude=set(excludes)).items() if v != []
-        }
+            for extra_attribute in self.get_extra_attributes():
+                if extra_attribute not in extended_attributes:
+                    excludes.append(extra_attribute)
+        data = {}
+        for k, v in self.model_dump(mode="json", exclude_none=False, exclude=set(excludes)).items():
+            if k not in extended_attributes:
+                if v == [] and k:
```

Comment:
Out of curiosity (or because I am too stupid after vacation) - why do we need the `and k` clause here but not on line 100? I.e. in which cases does `k` evaluate to false?

Reply:
I also don't see why we need `and k`. Good catch 👍